src/uid2Sdk.ts

```diff
@@ -329,10 +337,17 @@ export class UID2 {
 
     this._identity = validity.identity;
     if (validity.identity) {
-      this._cookieManager.setCookie(validity.identity);
+      if (this._opts.useCookie) {
+        this._cookieManager.setCookie(validity.identity);
+      }
+      else {
+        this._cookieManager.removeCookie();
```

Comment:
I like the idea of removing the cookie here, but we'll need to be very sure there aren't any cases where the cookie might be removed if we haven't loaded it properly yet - we don't want to remove a valid cookie if we aren't definitely going to store it.

Reply:
to be extra defensive, I could change `getValue` to a public method and do:
```
this._localStorageManager.setValue(validity.identity);
if (this._localStorageManager.getValue()) this._cookieManager.removeCookie();
```
Would that address your concern?

Reply:
Yeah, I think that's reasonable.

Reply:
done!